lib/spack/spack/package.py

```diff
@@ -1188,11 +1188,13 @@ def do_install(self,
         if self.spec.external:
             return self._process_external_package(explicit)
 
+        reuse_stage = not kwargs.get('restage', False)
```

Comment:
code clarity note: Let's rename this to "restage" and flip its semantics. Then in self.repair_partial we can say `if restage: # do restage` and we avoid any double negatives